examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+            entries1 = htemp->GetEntries();
+            nBins1 = htemp->GetNbinsX();
+            low1 = htemp->GetXaxis()->GetXmin();
+            high1 = htemp->GetXaxis()->GetXmax();
+        }
+
+        command1 = leafName1 + ">>hist1(" + nBins1 + ", " + low1 + ", " + high1 + ")";
+        //    cout << command1 << endl;
+        tree1->Draw(command1);
+        auto hist1 = (TH1F*)gPad->GetPrimitive("hist1");
+        int entries2{0};
+        float low2{0.};
+        float high2{0.};
+        int nBins2{0};
+        if (hist1) {
+            entries2 = hist1->GetEntries();
+            nBins2 = hist1->GetNbinsX();
+            low2 = hist1->GetXaxis()->GetXmin();
+            high2 = hist1->GetXaxis()->GetXmax();
+        }
+
+        if ((0 == entries1 && 0 != entries2) || (0 != entries1 && 0 == entries2)) {
+            std::cout << "One of the distributions is empty" << std::endl;
+            okay = kFALSE;
+        }
+        if (0 == entries1 && 0 == entries2) {
+            outstream << "Both Histograms are empty." << std::endl;
+
+            hist1->Clear();
+            htemp->Clear();
+            //      numTestedLeaves++;
+            //      numEmptyLeaves++;
+            continue;
+        }
+
+        // When executing the draw command "Leaf1 - Leaf2" the subtraction is
+        // executed entry by entry. If the content of the class members are
+        // identical the result is a histogram with a delta function at 0
+        // If the content is differnt one gets a distribution which is
+        // detected.
+        outstream << "Comparing " << leafName << " and " << leafName1 << std::endl;
+        TString command = leafName + "-" + leafName1 + ">>hist(20, -10.,10.)";
+        tree1->Draw(command);
+        auto hist = (TH1F*)gPad->GetPrimitive("hist");
+        numTestedLeaves++;
+
+        // Check if the entries in the tree are identical
+        Bool_t leafIsIdentical = CheckEntriesIdentical(hist, outstream);
+
+        // If the entries are not identical check if the histograms are
+        // identical. This is the case if the entries in the tree are sorted
+        // differently
+        if (leafIsIdentical) {
+            numIdenticalEntries++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking for identical histograms" << endl;
+            leafIsIdentical = CheckHistogramIdentical(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numIdenticalHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking Kolmogorov" << endl;
+            leafIsIdentical = CheckHistogramKolmogorov(htemp, hist1, outstream);
+        }
```

Comment:
```suggestion
        
        outstream << "Checking Kolmogorov" << endl;
        auto const kolmo = origHist->KolmogorovTest(newHist);
        outstream << "Result of Kolmogorov test: " << kolmo << endl;
        leafIsIdentical = kolmo > 0.99;
```

* no else block needed -> improve readability
* remove `CheckHistogramKolomogorov()` helper, it is only called once and does not add anything -> netto -12 lines of code

Reply:
removed the else block. Did not remove the helper function.

Reply:
I think, some helper functions are okay, if they reduce the size/complexity of the main thing.